src/bitcoin/poller/looper.rs

```diff
@@ -115,7 +115,8 @@ fn update_coins(
         .values()
         .chain(received.iter())
         .filter_map(|coin| {
-            if coin.spend_txid.is_none() {
+            // Always check for spends when the spend tx is not confirmed as it might get RBF'd.
+            if coin.spend_txid.is_none() || coin.spend_block.is_none() {
```

Comment:
Ok, I checked again that spend_txid can be set and not coin.spend_block
https://github.com/wizardsardine/liana/blob/e549510e218edce31a0deced9a348e7bf64e1d51/src/database/sqlite/mod.rs#L448